openshift/fluentd.bc.yaml

```diff
@@ -0,0 +1,94 @@
+---
+apiVersion: v1
+kind: Template
+labels:
+  build: ${APP_NAME}-${INSTANCE}-custom
+  template: ${APP_NAME}-${INSTANCE}-template-bc
+metadata:
+  name: ${APP_NAME}-${INSTANCE}-template-bc
+objects:
+  - apiVersion: v1
+    kind: ImageStream
+    metadata:
+      name: ${APP_NAME}-custom
+    spec:
+      lookupPolicy:
+        local: false
+
+  - apiVersion: v1
+    kind: BuildConfig
+    metadata:
+      name: ${APP_NAME}-${INSTANCE}-custom
+    spec:
+      completionDeadlineSeconds: 600
+      failedBuildsHistoryLimit: 3
+      nodeSelector:
+      output:
+        to:
+          kind: ImageStreamTag
+          name: "${APP_NAME}-custom:${INSTANCE}"
+      postCommit: {}
+      resources:
+       limits:
+          cpu: "${CPU_LIMIT}"
+          memory: "${MEMORY_LIMIT}"
+       requests:
+          cpu: "${CPU_REQUEST}"
+          memory: "${MEMORY_REQUEST}"
+      runPolicy: SerialLatestOnly
+      source:
+        type: Dockerfile
+        dockerfile: |-
+          FROM fluent/fluentd:v1.7-debian-1
+
+          # Use root account to use apt
+          USER root
+
+          # below RUN includes plugins:
+          # - fluent-plugin-rewrite-tag-filter
+          # - elasticsearch (in case we ever use it)
+
+          RUN buildDeps="sudo make gcc g++ libc-dev" \
+          && apt-get update \
+          && apt-get install -y --no-install-recommends $buildDeps \
+          && sudo gem install fluent-plugin-rewrite-tag-filter \
+          && sudo gem install fluent-plugin-elasticsearch \
+          && sudo gem sources --clear-all \
+          && SUDO_FORCE_REMOVE=yes \
+              apt-get purge -y --auto-remove \
+                            -o APT::AutoRemove::RecommendsImportant=false \
+                            $buildDeps \
+          && rm -rf /var/lib/apt/lists/* \
+          && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem
+
+          USER fluent
+      strategy:
+        type: Docker
+      successfulBuildsHistoryLimit: 3
+parameters:
+  - name: APP_NAME
+    description: Application name
+    displayName: Application name
+    required: true
+    value: fluentd
+  - name: INSTANCE
+    description: The deployment instance name
+    displayName: Deployment Instance
+    required: true
+    value: master
+  - name: CPU_LIMIT
+    description: Limit Peak CPU per pod (in millicores ex. 1000m)
+    displayName: CPU Limit
+    value: 2000m
```

Comment:
Since this is a build config with a `spec.completionDeadlineSeconds` of 600 seconds, you can bump this value up even higher if the act of building this is CPU bound (we're using the time-limited queue already so it's ok to request more resources in the short term if necessary... same applies for memory if required)